command/deployment_status.go

```diff
@@ -358,3 +391,247 @@ func formatDeploymentGroups(d *api.Deployment, uuidLength int) string {
+					return
+				}
+
+				// Check for noop/no target rollbacks
+				if rollback.ID == deploy.ID {
+					break UPDATE
+				}
+
+				d.Close()
+				c.ttyMonitor(client, rollback.ID, index, verbose)
+				return
+			} else {
+				break UPDATE
+			}
+		case structs.DeploymentStatusSuccessful, structs.DeploymentStatusCancelled, structs.DeploymentStatusDescriptionBlocked:
+			break UPDATE
+		default:
+			q.WaitIndex = meta.LastIndex
+			continue
+		}
+	}
+	// Render one final time with completion message
+	d.Set(endSpinner, statusComponent)
+	d.RenderFrame()
+}
+
+// Used for Windows and non-tty
+func (c *DeploymentStatusCommand) defaultMonitor(client *api.Client, deployID string, index uint64, verbose bool) {
+	writer := uilive.New()
+	writer.Start()
+	defer writer.Stop()
+
+	var length int
+	if verbose {
+		length = fullId
+	} else {
+		length = shortId
+	}
+
+	q := api.QueryOptions{
+		AllowStale: true,
+		WaitIndex:  index,
+		WaitTime:   2 * time.Second,
+	}
+
+	for {
+		deploy, meta, err := client.Deployments().Info(deployID, &q)
+		if err != nil {
+			c.Ui.Error(c.Colorize().Color(fmt.Sprintf("%s: Error fetching deployment", formatTime(time.Now()))))
+			return
+		}
+
+		status := deploy.Status
+		info := formatTime(time.Now())
+		info += fmt.Sprintf("\n%s", formatDeployment(client, deploy, length))
+
+		if verbose {
+			info += "\n\n[bold]Allocations[reset]\n"
+			allocs, _, err := client.Deployments().Allocations(deployID, nil)
+			if err != nil {
+				info += "Error fetching allocations"
+			} else {
+				info += formatAllocListStubs(allocs, verbose, length)
+			}
+		}
+
+		// Add newline before output to avoid prefix indentation when called from job run
+		msg := c.Colorize().Color(fmt.Sprintf("\n%s", info))
+
+		// Print in place if tty
+		_, isStdoutTerminal := term.GetFdInfo(os.Stdout)
+		if isStdoutTerminal {
+			fmt.Fprint(writer, msg)
+		} else {
+			c.Ui.Output(msg)
+		}
+
+		switch status {
+		case structs.DeploymentStatusFailed:
+			if hasAutoRevert(deploy) {
+				// Wait for rollback to launch
+				time.Sleep(1 * time.Second)
+				rollback, _, err := client.Jobs().LatestDeployment(deploy.JobID, nil)
+
+				// Separate rollback monitoring from failed deployment
+				// Needs to be after time.Sleep or it messes up the formatting
+				c.Ui.Output("")
+				if err != nil {
+					c.Ui.Error(c.Colorize().Color(
+						fmt.Sprintf("%s: Error fetching deployment of previous job version", formatTime(time.Now())),
+					))
+					return
+				}
+
+				// Check for noop/no target rollbacks
+				if rollback.ID == deploy.ID {
+					return
+				}
+				c.defaultMonitor(client, rollback.ID, index, verbose)
+			}
+			return
+
+		case structs.DeploymentStatusSuccessful, structs.DeploymentStatusCancelled, structs.DeploymentStatusDescriptionBlocked:
+			return
+		default:
+			q.WaitIndex = meta.LastIndex
+			continue
+		}
+	}
+}
+
+func (c *DeploymentStatusCommand) monitor(client *api.Client, deployID string, index uint64, verbose bool) {
```

Comment:
This is a tiny language style nit, but typically with go you'll want to order methods so that the definition is as close to the call site as possible. (This is in contrast to something like Python or Bash where with free functions you _can't_ call before the definition but with methods you can.)

So I'd order this `Run`, then `monitor`, then `ttyMonitor`, then `defaultMonitor`. (We're definitely not consistent about this in the code base, but it's nice to have for new code!)